sources/tools/Stride.Importer.Assimp/MeshConverter.cs

```diff
@@ -0,0 +1,1375 @@
+            BuildLayeredSurface(pMaterial, false, false, dummyColor.ToStrideColor(), 0.0f, TextureType.TextureTypeNormals, finalMaterial);
+            BuildLayeredSurface(pMaterial, false, false, dummyColor.ToStrideColor(), 0.0f, TextureType.TextureTypeDisplacement, finalMaterial);
+            BuildLayeredSurface(pMaterial, hasAmbientColor, false, ambientColor.ToStrideColor(), 0.0f, TextureType.TextureTypeAmbient, finalMaterial);
+            BuildLayeredSurface(pMaterial, false, hasOpacity, dummyColor.ToStrideColor(), opacity, TextureType.TextureTypeOpacity, finalMaterial);
+            BuildLayeredSurface(pMaterial, false, hasSpecPower, dummyColor.ToStrideColor(), specPower, TextureType.TextureTypeShininess, finalMaterial);
+            BuildLayeredSurface(pMaterial, hasEmissiveColor, false, emissiveColor.ToStrideColor(), 0.0f, TextureType.TextureTypeEmissive, finalMaterial);
+            BuildLayeredSurface(pMaterial, false, false, dummyColor.ToStrideColor(), 0.0f, TextureType.TextureTypeHeight, finalMaterial);
+            BuildLayeredSurface(pMaterial, hasReflectiveColor, false, reflectiveColor.ToStrideColor(), 0.0f, TextureType.TextureTypeReflection, finalMaterial);
+
+            return finalMaterial;
+        }
+
+        private unsafe void SetMaterialColorFlag(Silk.NET.Assimp.Material* pMaterial, string materialColorBase, ref bool hasMatColor, ref System.Numerics.Vector4 matColor, bool condition)
+        {
+            if (assimp.GetMaterialColor(pMaterial, materialColorBase, 0, 0, ref matColor) == Return.ReturnSuccess && condition)
+            {
+                hasMatColor = true;
+            }
+        }
+        private unsafe void SetMaterialFloatArrayFlag(Silk.NET.Assimp.Material* pMaterial, string materialBase, ref bool hasMatProperty, float matColor, bool condition)
+        {
+            if(assimp.GetMaterialFloatArray(pMaterial, materialBase, 0, 0, &matColor, (uint*)0x0) == Return.ReturnSuccess && condition)
+            {
+                hasMatProperty = true;
+            }
+        }
+
+        private bool IsNotBlackColor(System.Numerics.Vector4 diffColor)
+        {
+            return diffColor != System.Numerics.Vector4.Zero;
+        }
+
+        private unsafe void BuildLayeredSurface(Silk.NET.Assimp.Material* pMat, bool hasBaseColor, bool hasBaseValue, Color4 baseColor, float baseValue, TextureType textureType, MaterialAsset finalMaterial)
+        {
+            var nbTextures = assimp.GetMaterialTextureCount(pMat, textureType);
+
+            IComputeColor computeColorNode = null;
+            int textureCount = 0;
+            if (nbTextures == 0)
+            {
+                if (hasBaseColor)
+                {
+                    computeColorNode = new ComputeColor(baseColor);
+                }
+                //else if (hasBaseValue)
+                //{
+                //	computeColorNode = gcnew MaterialFloatComputeNode(baseValue);
+                //}
```

Comment:
Why is this commented out?

Reply:
I think that's commented C++/CLI code, since C# doesn't have a `gcnew` keyword. `gcnew` is specific to C++/CLI and C++/CX.

Reply:
`gcnew` in C++/CLI is just the `new` keyword in C#, i.e. instantiating a new instance managed by the Garbage Collector.

Reply:
Yes, but this is a C# file, so there's no use for it. Maybe used as some sort of reference?

Reply:
Right now I'm not very versed in the material system. Maybe @tebjan knows better, but if there is a `MaterialFloatComputeNode` (or there were one) that maybe is a `IComputeValue` (does that type still exist?) it is/was useful somehow.
Or maybe it is merged into `ComputeColor` at this point and we should just create a regular `ComputeColor` with a single scalar value?
Anyway, that `else if (hasBaseValue)` indicates there was an use for it.

Reply:
Then must be a half-implemented feature, or a remnant from some old code. As I said, I'm not very knowledgeable about the material system, so don't know what would be the uses of that even if it sounds useful somehow.

Reply:
I found original commit. Maybe It will be usefull for someone in the future: https://github.com/SiliconStudio/xenko//commit/d5ee29ea8efeceaa6eaf777ff903e4514f28c6f7